xm-commons-cache/src/main/java/com/icthh/xm/commons/cache/service/TenantAwareCacheManager.java

```diff
@@ -0,0 +1,73 @@
+package com.icthh.xm.commons.cache.service;
+
+import com.icthh.xm.commons.cache.TenantCacheManager;
+import com.icthh.xm.commons.tenant.TenantContextHolder;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TenantAwareCacheManager implements TenantCacheManager {
+
+    private final CacheManager delegate;
+
+    private final TenantContextHolder tenantContextHolder;
+
+    /**
+     * Constructor.
+     *
+     * @param delegate
+     *            the {@link CacheManager} to delegate to.
+     * @param tenantContextHolder
+     *            context to retrieve the current tenant
+     */
+    public TenantAwareCacheManager(final CacheManager delegate, final TenantContextHolder tenantContextHolder) {
+        this.tenantContextHolder = tenantContextHolder;
+        this.delegate = delegate;
+    }
+
+    @Override
+    public Cache getCache(final String name) {
+        String currentTenant = tenantContextHolder.getTenantKey();
+        if (isTenantInvalid(currentTenant)) {
```

Comment:
can we incapsulate this logic, and add tenant name to cache name inside?

` dynamicCaffeineCacheManager.getCache(TenantCacheManager.buildKey("TEST", "tcache"))`
>>>
` dynamicCaffeineCacheManager.getCache("tcache")`